Sources/StreamChat/ChatClient.swift

```diff
@@ -373,6 +373,16 @@ public class ChatClient {
         }
         userConnectionProvider = nil
     }
+    
+    /// Disconnects the chat client form the chat servers and removes all the local data related.
+    ///
+    /// - Parameters:
+    ///   - completion: The completion. Will be called when the logout is completed successfully.
+    ///                 If request fails, the completion will be called with an error.
+    public func logout(completion: @escaping ((Error?) -> Void)) {
+        disconnect()
+        databaseContainer.removeAllData(force: true, completion: completion)
+    }
```

Comment:
Do we really need a completion block here? 🤔 If we do, better to provide a default to `nil` as most likely, customers, won't need to wait for this. And also, how could the logout error? 🤔 

CC @polqf

Reply:
I added it mainly because in this case the `removeAllData` could fail and probably is a good idea that the implementers have feedback on that if they want to act depending on the error, but I also agree that a logout might be better without parameters and we address the completion internally by logging for instance similar to what I added on the `StreamChatWrapper` when using the `logout()` method. The default to `nil` approach might be a good idea too, but probably is better to remove the parameters altogether.

Reply:
I think it is essential for us to guide our customers to use our APIs in the correct way. The message we send if we don't add a completion block is wrong. We might find customers calling logout and just after trying to log in. This will end up in a race condition where the DB is still being cleared while we are trying to perform additions to the DB

Reply:
I find that a bit difficult, especially because the user will then need to log in, which takes quite some time. I'm fine with adding a completion block, but from a developer perspective is weird. Especially the error. How do you react to a failed logout?

Reply:
IMO the completion block is not necessary. In real use cases, there’s no case a user could log in as fast as the completion block. If we think it is necessary and some customers still have issues, we can add the completion block later